spyderlib/utils/introspection/rope_plugin.py

```diff
@@ -239,6 +239,25 @@ def set_pref(self, key, value):
         if self.project is not None:
             self.project.prefs.set(key, value)
 
+    def rename(self, code, position, new_name):
+        """
+        Returns code with identifier renamed using Rope
+        code (str): old source code
+        position (int): position of identifier to be renamed
+        new_name (str): new name of identifier
+        """
+        import os
+        temp_filename = 'ropefile-{}.py'.format(os.getpid())
+        temp_module = self.project.root.create_file(temp_filename)
+        temp_module.write(code)
+        from rope.refactor.rename import Rename
+        renamer = Rename(self.project, temp_module, position)
+        changes = renamer.get_changes(new_name)
+        self.project.do(changes)
+        result = temp_module.read()
+        temp_module.remove()
+        return result

```

Comment:
It would be really better to not use a temporary file. I tried to find an alternative but I couldn't find anything quickly so I don't know.
If a temporary file has to be used, it should use a context manager to be sure to remove the file even in case of error, and the name should be more unique that using the pid. See https://docs.python.org/3.5/library/tempfile.html#examples for example.